MovingBlocks/Terasology
# doc: fix broken links

### Contains

* moves files from docs-pre-merge to more reasonable places
* fixes broken links in README.md
* links knowledge base (migrated wiki) directly instead of dedicated README section (explanatory reference added to "contributing" README section)
* also improves existing non-broken links by using https instead of http

### How to test

Click on all links in https://github.com/MovingBlocks/Terasology/blob/doc/fix-broken-links/README.md and verify they are not broken.

Fixes #5166
